common/esxi_get_guest_config_options.yml

```diff
@@ -0,0 +1,312 @@
+        content: text
+      register: guest_supported_usb_ctrls
+      ignore_errors: True
+
+    - name: "Set supported USB controllers for guest id {{ guest_id }}"
+      set_fact:
+        guest_config_options: >-
+          {{ 
+            guest_config_options
+            | combine({
+                 'support_usb_controller': 
+                 guest_supported_usb_ctrls.matches | map(attribute='e') | map('extract', vm_device_types) | list 
+              })
+           }}
+      when:
+        - guest_supported_usb_ctrls is defined
+        - guest_supported_usb_ctrls.matches is defined
+        - guest_supported_usb_ctrls.matches | length > 0
+
+    - name: "Combine guest_id and hardware version into guest_config_options"
+      set_fact:
+        guest_config_options: >-
+          {{ 
+            guest_config_options
+            | combine({
+                 'guest_id': guest_id,
+                 'hardware_version': 'vmx-' ~  esxi_hardware_version
+                 })
+           }}
+
+    - name: "Remove temporary config option file"
+      file:
+        path: "{{ tmp_config_option_file }}"
+        state: absent
+  when: >
+    (get_config_options_result is undefined) or
+    (get_config_options_result.instance is undefined) or
+    (get_config_options_result.instance["Recommended config options"]) is undefined
+
+- name: "Set default CPU number for VM with guest id {{ guest_id }}"
+  set_fact:
+    guest_config_options: "{{ guest_config_options | combine({'default_cpu_number': (guest_config_options.default_cpu_cores_per_socket | int) * (guest_config_options.default_cpu_socket | int) }) }}"
+  when:
+    - guest_config_options.default_cpu_cores_per_socket is defined
+    - guest_config_options.default_cpu_cores_per_socket | int >= 1
+    - guest_config_options.default_cpu_socket is defined
+    - guest_config_options.default_cpu_socket | int >= 1
+
+- name: "Set default CPU number for guest id {{ guest_id }}"
+  set_fact:
+    guest_config_options: "{{ guest_config_options | combine({'default_cpu_number': 1}) }}"
+  when: >
+    (guest_config_options.default_cpu_number is undefined) or
+    (not guest_config_options.default_cpu_number)
+
+- name: "Set default USB controller for guest id {{ guest_id }}"
+  set_fact:
+    guest_config_options: "{{ guest_config_options | combine({'default_usb_controller': 'usb2'}) }}"
+  when: >
+    (guest_config_options.default_usb_controller is undefined) or
+    (not guest_config_options.default_usb_controller)
+
+- name: "Set default CDROM controller for guest id {{ guest_id }}"
+  set_fact:
+    guest_config_options: "{{ guest_config_options | combine({'default_cdrom_controller': 'ide'}) }}"
+  when: >
+    (guest_config_options.default_cdrom_controller is undefined) or
+    (not guest_config_options.default_cdrom_controller)
+
+- name: "Print guest id {{ guest_id }} config options on hardware version {{ esxi_hardware_version }}"
+  debug: var=guest_config_options
+
+# Example output:
```

Comment:
Example output can be put at the beginning like in other tasks.

Reply:
Moved example output at the beginning.